lib/rubies/game.rb

```diff
@@ -155,6 +165,45 @@ def check_answer(current, input, target)
       end
     end
 
+    # deletes the target from the current data structure
+    def deep_delete(current, target)
+      return current unless current.is_a? Enumerable
+      current.
+          reject  {|value| value == target}.
+          map     {|value| deep_delete(value, target)}
+    end
+
+    # creates an array of values for which any given answer should fail
+    def tests(current, target)
+      [ nil,
+        Array.new,
+        Hash.new,
+        RandomDataStructure.new.generate,
+        deep_delete(current, target)
+      ]
+    end
+
+    # test user input against various tests, for each of which the
+    # input should fail. Returns true only if the user input fails
+    # each of the tests.
+    def test_answer(current, input, target)
+      tests(current, target).all? do |test|
+        !run_test(test, input, target)
+      end
+    end
+
+    # test whether the answer provided by the given 
+    # routine matches the target answer
+    def run_test(current, input, target)
```

Comment:
Unused method argument - `current`. If it's necessary, use `_` or `_current` as an argument name to indicate that it won't be used.